frontend/app/[team]/apps/[app]/environments/[environment]/[[...path]]/page.tsx

```diff
@@ -798,10 +968,66 @@ export default function EnvironmentPath({
               <FolderBreadcrumbLinks path={params.path} />
             </div>
             {unsavedChanges && (
+              <div onClick={handleChangesReveal}>
               <Alert variant="warning" icon={true} size="sm">
-                You have undeployed changes to this environment.
+                <span className="mr-[-0.5rem]" onClick={handleChangesReveal}>
+                  You have undeployed changes to this environment.
+                </span>
+                <GenericDialog
+                  buttonVariant=""
+                  title="Undeployed changes"
+                  onClose={() => setIsChangesModalOpen(false)}
+                  buttonContent={<FaInfoCircle />}
+                >
+                  <div className="flex flex-col space-y-2">
```

Comment:
```suggestion
                  <div className="flex flex-col space-y-2 max-h-[85vh] overflow-auto">
```
Add a max height to handle overflow where there are a large number of changes